pkg/controller/daemon/daemoncontroller.go

```diff
@@ -659,6 +678,14 @@ func (dsc *DaemonSetsController) updateDaemonSetStatus(ds *extensions.DaemonSet)
 				sort.Sort(podByCreationTimestamp(daemonPods))
 				if v1.IsPodReady(daemonPods[0]) {
 					numberReady++
+					ready = true
+				}
+				if util.IsPodUpdated(dsPodTemplateSpecHash, daemonPods[0]) {
+					updatedNumberScheduled++
+					updated = true
+				}
+				if ready && updated {
+					numberAvailable++
```

Comment:
Yes, ref https://github.com/kubernetes/community/blob/master/contributors/design-proposals/daemonset-update.md#daemonset-controller:

- `.status.numberAvailable` = the total number of DaemonSet pods that have become `Ready` for `MinReadySeconds`
- `.status.numberUnavailable` = `.status.desiredNumberScheduled` - `.status.numberAvailable`